### ISSUE TITLE
Backpropagation of RE loss function

### ISSUE THREAD

**Issue by ts-kim** [ChessQian/cephaNN#1]
Hi, ChessQian.

I am interested in your paper CephaNN.
When reading your paper, I have a question about your loss function, RE loss,

I wonder how the proposed loss function can be backpropagated. 

I think that the additional part to the MSE loss is calculated with the "mask"s, E_G and E_H.
In my experience, the masking with condition, E=I(H(x,y)>t), is not differentiable, leading to the zero-gradients(no gradients) to the model.
Therefore, the model is updated with the MSE loss only.

If there is something I missed, please let me know.

Thank you.
